zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -211,27 +241,153 @@ object JSONParser {
         s => if (s.isEmpty) None else Some(s)
       ) and
-      (JsPath \ "group").formatNullable[GroupStatement] and
-      (JsPath \ "select").formatNullable[SelectStatement] and
-      (JsPath \ "global").formatNullable[GlobalAggregateStatement] and
+      (JsPath \ "group").formatNullable[UnresolvedGroupStatement] and
+      (JsPath \ "select").formatNullable[UnresolvedSelectStatement] and
+      (JsPath \ "global").formatNullable[UnresolvedGlobalAggregateStatement] and
       (JsPath \ "estimable").formatNullable[Boolean].inmap[Boolean](
         o => o.getOrElse(false),
         s => if (s) Some(s) else None
       )
-    ) (Query.apply, unlift(Query.unapply))
+    ) (UnresolvedQuery.apply, unlift(UnresolvedQuery.unapply))
+
+
+  def toUnresolved(query: Query): UnresolvedQuery =
+    UnresolvedQuery(
+      query.dataset,
+      query.lookup.map(toUnresolved(_)),
+      query.filter.map(toUnresolved(_)),
+      query.unnest.map(toUnresolved(_)),
+      query.groups.map(toUnresolved(_)),
+      query.select.map(toUnresolved(_)),
+      query.globalAggr.map(toUnresolved(_)),
+      query.isEstimable
+    )
+
+  def toUnresolved(lookup: LookupStatement): UnresolvedLookupStatement =
+    UnresolvedLookupStatement(
+      lookup.sourceKeys.map(_.name),
+      lookup.dataset,
+      lookup.lookupKeys.map(_.name),
+      lookup.selectValues.map(_.name),
+      lookup.as.map(_.name)
+    )
+
+  def toUnresolved(aggr: AggregateStatement): UnresolvedAggregateStatement =
+    UnresolvedAggregateStatement(
+      aggr.field.name,
+      aggr.func,
+      aggr.as.name
+    )
+
+  def toUnresolved(filter: FilterStatement): UnresolvedFilterStatement =
+    UnresolvedFilterStatement(
+      filter.field.name,
+      filter.funcOpt,
+      filter.relation,
+      filter.values
+    )
+
+  def toUnresolved(unnest: UnnestStatement): UnresolvedUnnestStatement =
+    UnresolvedUnnestStatement(
+      unnest.field.name,
+      unnest.as.name
+    )
+
+  def toUnresolved(by: ByStatement): UnresolvedByStatement =
+    UnresolvedByStatement(
+      by.field.name,
+      by.funcOpt,
+      by.as.map(_.name)
+    )
+
+  def toUnresolved(group: GroupStatement): UnresolvedGroupStatement =
+    UnresolvedGroupStatement(
+      group.bys.map(toUnresolved(_)),
+      group.aggregates.map(toUnresolved(_))
+    )
+
+
+  def toUnresolved(globalAggr: GlobalAggregateStatement): UnresolvedGlobalAggregateStatement =
+    UnresolvedGlobalAggregateStatement(
+      toUnresolved(globalAggr.aggregate)
+    )
+
+  def toUnresolved(select: SelectStatement): UnresolvedSelectStatement =
+    UnresolvedSelectStatement(
+      select.orderOn.zip(select.order).map {
+        case (field, order) =>
+          order match {
+            case SortOrder.DSC => "-" + field.name
+            case SortOrder.ASC => field.name
+          }
+      },
+      select.limit,
+      select.offset,
+      select.fields.map(_.name)
+
+    )
 
 }
+
+
+case class UnresolvedQuery(dataset: String,
```

Comment:
should we also put all these `Unresolved` objects and `toUnresolved` functions to a separate unified `object Unresolved`? it will be more clear. Because the `unresolved` will be only used once. The `resolved` ones are used all the time, people will go back and forth to view the object and change it. we'd better keep that core part clean.